### PR TITLE
added: Buffer.loadChannelDialog and its corresponding help documentation

### PR DESCRIPTION
**Pull request by prko**
## Purpose and Motivation
It would be convenient for SuperCollider to provide a counterpart for `Buffer.readChannel`, as it already provides `Buffer.loadDialog` as the counterpart to `Buffer.read`.
   - A new class method for Buffer is added: `Buffer.loadChannelDialog`.
   - The appropriate part for the new method above to `Buffer.schelp` file is added.
   - Some text lines are arranged to avoid wrong automatic indentation.

## Types of changes

- Documentation
- New feature

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

**Review by cdbzb — COMMENTED** [review]
Hi @prko thanks for this! I support the idea.

I do have a comment regarding the examples in the documentation. 

My understanding is that code blocks should *not* be indented: ie

```
(
a = 3; \\right
)
```
not
```
(
    a = 3; //wrong!
)
```

so these should be reverted!

**Comment by prko** [comment]
@cdbzb I did it! Thanks!

**Comment by cdbzb** [comment]
Wow that was fast! thank you!

while we're here: I think this (and similar!)

```
		^super.newCopyArgs(server,
			bufnum,
			numFrames,
			numChannels).sampleRate_(server.sampleRate).cache
```
would be better
```
		^super.newCopyArgs(
			server,
			bufnum,
			numFrames,
			numChannels
		).sampleRate_(server.sampleRate).cache
```

and from my understanding of the code style guidelines https://github.com/supercollider/supercollider/wiki/Code-style-guidelines#method-calls

```
		^super.newCopyArgs(server, bufnum)
		.doOnInfo_(action).cache
		.allocReadChannel(path, startFrame, numFrames, channels,
```

should be indented like

```
		^super.newCopyArgs(server, bufnum)
			.doOnInfo_(action).cache
			.allocReadChannel(path, startFrame, numFrames, channels,
```

**Comment by cdbzb** [comment]
I'd also suggest that you save the changes in the Line Layout Help for a different PR - its better to segregate PRs by purpose.

...or better you might separate it into one *functional* PR (adding a method) which requires testing etc. and another PR for various reformatting and cleanup efforts. It makes it easier for us to test and review, and also makes the history easier to understand if that makes sense.

**Comment by cdbzb** [comment]
looking forward to testing the actual method! but that's it for my day today - more tomorrow!

**Comment by prko** [comment]
@cdbzb  I changed the parts in the part `Buffer.sc` according to your suggestion, and reverted the file `LineLayout.schelp`. I will make a new PR "increased readability and comfortable code evaluation" later with other files. I am not sure when I can do it.

File: HelpSource/Classes/Buffer.schelp [hunk 1]
```diff
@@ -358,7 +358,32 @@ b = Buffer.loadDialog(s, action: { arg buffer;
 )
 x.free; b.free;
 ::
-
+		
+method:: loadChannelDialog
+As link::#*readChannel:: above, but gives you a load dialog window to browse for a file. Cocoa compatible.
```
Comment (telephon) [inline HelpSource/Classes/Buffer.schelp:363]: you could drop "Cocoa compatible."? – it should be compatible to all platforms that sclang supports.

**Review by telephon — APPROVED** [review]
Since this matches the second file reading operation and simple enough, I think it is justified.

**Review by cdbzb — COMMENTED** [review]
Tetsted on Mac - works correctly

File: SCClassLibrary/Common/Control/Buffer.sc [hunk 2]
```diff
@@ -186,11 +189,11 @@ Buffer {
 					sndfile.writeData(collection);
 					sndfile.close;
 					^super.newCopyArgs(server, bufnum)
-						.cache.doOnInfo_({ |buf|
-							if(File.delete(path), { buf.path = nil},
-								{("Could not delete data file:" + path).warn;});
-							action.value(buf);
-						}).allocRead(path, 0, -1, {|buf| ["/b_query", buf.bufnum] })
+					.cache.doOnInfo_({ |buf|
```
[the remark this answers is not in the document]
Comment (prko) [reply SCClassLibrary/Common/Control/Buffer.sc]: done!

File: SCClassLibrary/Common/Control/Buffer.sc [hunk 3]
```diff
@@ -88,35 +91,35 @@ Buffer {
...
 	}
 
 	*readChannel { arg server, path, startFrame = 0, numFrames = -1, channels, action, bufnum;
 		server = server ? Server.default;
 		bufnum ?? { bufnum = server.nextBufferNumber(1) };
 		^super.newCopyArgs(server, bufnum)
-					.doOnInfo_(action).cache
-					.allocReadChannel(path, startFrame, numFrames, channels,
-						{|buf|["/b_query", buf.bufnum]})
+		.doOnInfo_(action).cache
```
[the remark this answers is not in the document]
Comment (prko) [reply SCClassLibrary/Common/Control/Buffer.sc]: done!

File: HelpSource/Classes/Buffer.schelp (hunk 1 above)
Comment (prko) [reply HelpSource/Classes/Buffer.schelp]: done! Also removed the corresponding part in `.loadDialog`

**Comment by prko** [comment]
I have tested the code block in the help document and it worked on MacOS 13.6.3 (silicon), Windows 11 Pro (arm64) and Ubuntu 22.04 (arm64).

File: SCClassLibrary/Common/Control/Buffer.sc [hunk 4]
```diff
@@ -673,7 +675,20 @@ Buffer {
 		buffer = super.newCopyArgs(server, bufnum).cache;
 		Dialog.openPanel({ arg path;
 			buffer.doOnInfo_(action)
-				.allocRead(path, startFrame, numFrames, { ["/b_query", buffer.bufnum] })
+			.allocRead(path, startFrame, numFrames, { ["/b_query", buffer.bufnum] })
```
Comment (cdbzb) [inline SCClassLibrary/Common/Control/Buffer.sc:678]: sorry @prko I missed one! this one should be indented also - then we're good to go!

Comment (prko) [reply SCClassLibrary/Common/Control/Buffer.sc]: Done, I hope I correctly understood you

**Review by cdbzb — APPROVED** [review]
thanks!

**Pull request merged**